lib/CBattleCallback.cpp

```diff
@@ -1020,8 +1020,28 @@ TDmgRange CBattleInfoCallback::calculateDmgRange(const BattleAttackInfo &info) c
 		multBonus *= 0.5;
 	}
 
-	// TODO attack on petrified unit 50%
 	// blinded unit retaliates
+	auto blinded = info.attackerBonuses->getBonus(Selector::source(Bonus::SPELL_EFFECT, SpellID::BLIND));

```

Comment:
blind spell has 3 effects = 3 bonuses (2 of them with 0 val). Which bonus will you get?